lib/Auth/Process/DynamicAssurance.php

```diff
@@ -264,12 +273,20 @@ public function process(&$state)
             "[DynamicAssurance][process] Assurance Values: " . var_export($assurance_from_candidates, true)
         );
 
+        // Check the required Assurance values
+        $append_default = true;
+        if(!empty($this->minAssurance)) {
+            $found_values = array_intersect($assurance_from_candidates, $this->minAssurance);
+            $append_default = !empty($found_values) ? false : true;
+        }
+
         // Append the Default Assurance if the Assurance list is empty
-        if(!empty($this->defaultAssurance)
-           && empty($assurance_from_candidates)) {
+        if((!empty($this->defaultAssurance) && $append_default)
+           || empty($assurance_from_candidates))
+        {
```

Comment:
what happens `if (empty(assurance_from_cadidates) && empty($this->defaultAssurance))` ?

Reply:
Currently we will get no eduPersonAssurance attribute in the state